sunpy/net/dataretriever/tests/test_goes_ud.py

```diff
@@ -15,9 +19,72 @@
 from sunpy.time import parse_time, is_time_equal
 
 
-@pytest.fixture
-def LCClient():
-    return goes.XRSClient()
+LCClient = goes.XRSClient()
+
+
+def mock_query_object(start_date, end_date):
+    """
+    Creation of a QueryResponse object, and prefill some
+    downloaded data from goes.XRSClient().fetch(Time('20 ..)
+    """
+    # Create a mock QueryResponse object
+    map_ = {
+        'TimeRange' : TimeRange(parse_time(start_date), parse_time(end_date)),
+        'Time_start': parse_time(start_date),
+        'Time_end':  parse_time(end_date),
+        'source': 'nasa',
+        'instrument': 'goes',
+        'physobs': 'irradiance',
+        'provider': 'sdac'
+    }
+
+    st_datetime = datetime.strptime(start_date, '%Y/%m/%d')
+    ed_datetime = datetime.strptime(end_date, '%Y/%m/%d') + timedelta(days=1)
+    time_range = TimeRange(st_datetime, ed_datetime).split(int((ed_datetime-st_datetime).days))
+
+    resp = QueryResponse.create(map_,
+                                LCClient._get_url_for_timerange(
+                                    map_['TimeRange']),
+                                time=time_range)
+    # Attach the client with the QueryResponse
+    resp.client = LCClient
+    return resp
+
+
+@pytest.mark.remote_data
+def test_fetch_working():
+    """
+    Tests if the online server for noaa is working.
+    Uses the url : ftp://ftp.swpc.noaa.gov/pub/weekly/RecentIndices.txt
+    """
+    qr1 = LCClient.search(Time('1995/06/03', '1995/06/05'),
+                          Instrument('XRS'))
+
+    # Mock QueryResponse object
+    mock_qr = mock_query_object('1995/06/03', '1995/06/05')
+
+    # Compare if two objects have the same attribute
+
+    mock_qr = mock_qr[0]
+    qr = qr1[0]
+
+    assert mock_qr.source == qr.source
+    assert mock_qr.provider == qr.provider
+    assert mock_qr.physobs == qr.physobs
+    assert mock_qr.instrument == qr.instrument
+    assert mock_qr.url == qr.url
+
+    # Here the mock object contains `datetime.datetime` object by default,
+    # so no parsing required
+
+    sdate1 = mock_qr.time.start.value
+    sdate2 = datetime.strptime(qr.time.start.value, "%Y-%m-%dT%H:%M:%S.%f")
+
+    edate1 = mock_qr.time.end.value
+    edate2 = datetime.strptime(qr.time.end.value, "%Y-%m-%dT%H:%M:%S.%f")
+
+    assert abs(sdate2 - sdate1) < timedelta(seconds=1)
+    assert abs(edate2 - edate1) < timedelta(seconds=1)
```

Comment:
You should be using `sunpy.time.astropy_time._is_time_equal` to check the equality of the `Time` objects.

Reply:
Shouldn't we refrain from using private methods?

Reply:
This function is made for use in tests exclusively. There might be rounding errors when comparing astropy times.

Reply:
Okay, that should help me replace a crude method.

Reply:
Also there should be nothing wrong with using private sunpy functions. It is using upstream ones that might  be a problem down the road.

Reply:
Are you sure that this method exists, for I am getting this error - `ModuleNotFoundError: No module named 'sunpy.time.astropy_time'
`

Reply:
The function was moved into `sunpy.time.time` if I recall and the private name was dropped.

Reply:
Okay, so I should stick to my original convention.

Reply:
> The function was moved into sunpy.time.time if I recall and the private name was dropped.

0_0

Reply:
I found out that `TimeRange` cannot be subtracted, so I am going to drop this idea.